lua/neogen/snippet.lua

```diff
@@ -52,13 +52,45 @@ end
 ---@param pos table a tuple of row, col
 ---@private
 snippet.engines.luasnip = function(snip, pos)
-    local ok, luasnip = pcall(require, "luasnip")
+    local ok, ls = pcall(require, "luasnip")
     if not ok then
         notify("Luasnip not found, aborting...", vim.log.levels.ERROR)
         return
     end
+
+    local types = require("luasnip.util.types")
+
+    -- Append a new line to create the snippet
     vim.fn.append(pos[1], "")
-    luasnip.lsp_expand(table.concat(snip, "\n"), { pos = { pos[1], pos[2] } })
+
+    -- Convert the snippet to string
+    local _snip = table.concat(snip, "\n")
+
+    ls.snip_expand(
+        ls.s("", ls.parser.parse_snippet(nil, _snip), {
+
+            child_ext_opts = {
+                -- for highlighting the placeholders
+                [types.insertNode] = {
+                    -- when inside placeholder
+                    active = {
```

Comment:
`active` doesn't need to be set here explicitly, it inherits the values from `passive` (and `passive` in turn from `snippet_passive`) (just fyi, I'd understand having it there for clarity)

Reply:
Oh thanks ! Removed it.